src/js/utils/dom.js

```diff
@@ -181,6 +181,48 @@ export function createEl(tagName = 'div', properties = {}, attributes = {}, cont
   return el;
 }
 
+/**
+ * Determines if a control is visible to the user
+ *
+ * @param {HTMLElement} control - The control whose visibility is being checked
+ * @return {boolean} - True if the control is visible
+ */
+export function isVisible(control) {
+  if (!control) {
+    return false;
+  }
+
+  // Using `includes` below to catch cases where `!important` is used in the style
+  return !computedStyle(control, 'display').includes('none') && !computedStyle(control, 'visibility').includes('hidden');
```

Comment:
Unfortunately, we can't use `.includes` as we still support IE11 and do not require a shim to be included (pun intended).
This can probably be something like
```suggestion
  // Account for usage of `!important
  return !(/^none/).test(computedStyle(control, 'display')) && !(/^hidden/).test(computedStyle(control, 'visibility'));
```